lib/app.jsx

```diff
@@ -84,6 +87,28 @@ const isElectron = ( () => {
 	return () => foundElectron;
 } )();
 
+const when = ( condition ) => ( action, fallback = noop ) => ( ... args ) => (
+	condition( ... args ) ? action( ... args ) : fallback( ... args )
+)
+
+const preventDefaultThen = ( fn ) => ( e ) => {
+	e.preventDefault()
+	return fn( e )
+}
+
+const whenCanResize = when( ( { canResize } ) => canResize );
+const whenIsResizing = when( ( { isResizing } ) => isResizing );
+
+const detectCanResize = ( { onResizable, sourceListSelector } ) => ( e ) => {
+	const node = document.querySelector( sourceListSelector )
```

Comment:
With **React** we can pass a function in on a components `ref` prop and use that to eliminate the need for these things.

```js
updateDescription() {
	this.displayParagraph.innerHTML = 'Bazinga!;
}

render() {
	return (
		<div>
			<h1>Test</h1>
			<p ref={ node => ( this.displayParagraph = node ) }>Gobble gobble!</p>
		</div>
	);
}
```

I think we can eliminate the checks if `this.displayParagraph` exists (unless we might programmatically remove it) because it should be initialized when the component mounts. Maybe it won't be there at `componentWillMount`

Hope this helps!